docs/_docs/security/cache-encryption-key-rotation.adoc

```diff
@@ -0,0 +1,147 @@
+|===
+
+=== From code
+
+The cache encryption key change process can be managed programmatically:
+
+[tabs]
+--
+tab:Java[]
+
+[source, java]
+----
+include::{javaCodeDir}/TDE.java[tags=cache-group-key-rotation, indent=0]
+----
+--
+
+== Managing the re-encryption process
+
+Re-encrypting existing data can take a long time. This is a fault-tolerant operation that automatically continues after a node restart.
+The previous encryption key is automatically removed when all local partitions are encrypted with the new key, and the last link:persistence/native-persistence#write-ahead-log[Write-Ahead Log] segment, which may contain entries encrypted with the previous key, is removed from disk.
+
+NOTE: Re-encryption uses link:persistence/native-persistence#write-ahead-log[Write-Ahead Log] for physical recovery and may affect performance of cache operations.
+
+There are several options to manage the performance impact of re-encryption:
+
+* Limit the rate of re-encryption by using a configuration parameter or by using the CLI at runtime.
+* Temporarily suspend re-encryption using CLI command.
+
+Ignite 2.10 introduces a new configuration section `EncryptionConfiguration`, that is part of `DatastorageConfiguration`.
+[cols="1,1,1",opts="header"]
+|===
+|Property | Default value | Description
+|reencryptionRateLimit | 0 (unlimited) | Re-encryption rate limit in megabytes per second.
+|reencryptionBatchSize | 100 | The number of pages that is scanned during re-encryption under checkpoint lock.
+|===
+
+=== Using XML configuration to limit re-encryption rate
+[source, xml]
+----
+include::code-snippets/xml/tde.xml[tags=ignite-config;!discovery;!encryption;!cache;!discovery, indent=0]
+----
+
+=== Using CLI to control re-encryption
+
+The `control.sh|bat` script provides the ability to change the re-encryption rate, suspend and resume background re-encryption at runtime.
+
+NOTE: After the node restarts, the suspended background re-encryption will continue automatically, and the rate limit will be set from the local XML configuration (if any) or set to unlimited rate (default).
+
+[source,shell]
+----
+# View the re-encryption status of the cache group.
+control.sh|bat --encryption reencryption_status cacheGroupName
+
+# Suspend re-encryption of the cache group.
+control.sh|bat --encryption suspend_reencryption cacheGroupName
+
+# Resume (suspended) re-encryption of the cache group.
+control.sh|bat --encryption resume_reencryption cacheGroupName
+
+# View the re-encryption rate limit.
+control.sh|bat --encryption reencryption_rate_limit
+
+# Set the re-encryption rate limit to 2.5 MB/s.
+control.sh|bat --encryption reencryption_rate_limit 2.5
+
+# Set re-encryption to an unlimited rate.
+control.sh|bat --encryption reencryption_rate_limit 0
+----
+
+The re-encryption status can also be obtained using JMX metrics, which are described in the link:monitoring-metrics/new-metrics#cache-groups[Cache group metrics] section.
```

Comment:
'...using JMX metrics described in the link:monitoring-metrics/new-metrics#cache-groups[Cache group metrics] section.'